src/core/lib/transport/transport.h

```diff
@@ -228,6 +229,61 @@ struct CallArgs {
 using NextPromiseFactory =
     std::function<ArenaPromise<ServerMetadataHandle>(CallArgs)>;
 
+// TODO(ctiller): eventually drop this when we don't need to reference into
+// legacy promise calls anymore
+class CallSpineInterface {
+ public:
+  virtual ~CallSpineInterface() = default;
+  virtual Pipe<ClientMetadataHandle>& client_initial_metadata() = 0;
+  virtual Pipe<ServerMetadataHandle>& server_initial_metadata() = 0;
+  virtual Pipe<MessageHandle>& client_to_server_messages() = 0;
+  virtual Pipe<MessageHandle>& server_to_client_messages() = 0;
+  virtual Pipe<ServerMetadataHandle>& server_trailing_metadata() = 0;
+  GRPC_MUST_USE_RESULT virtual absl::nullopt_t Cancel(
```

Comment:
As discussed, please add a comment explaining why `GRPC_MUST_USE_RESULT` is being used with `absl::nullopt_t`, and maybe a TODO to find a more elegant way to accomplish the same thing.

Reply:
done